docs/ideas/yaml/jmespath.toml

```diff
@@ -0,0 +1,89 @@
+# The values below were taken from .github/workflows/python.yaml in this repo
+
+# 1. JMESPath as part of the section name, after the file name.
+# Everything after the file name is considered a JMESPath https://jmespath.org/
+# Format: ["path/to/file.ext".jmes.path.expression]
+#
+# 2. "jobs.build.strategy.matrix" should have "os" and "python-version"
+# 3. Both are lists, and they have to be exactly as described here.
+[".github/workflows/python.yaml".jobs.build.strategy.matrix]
+os = ["ubuntu-latest", "macos-latest", "windows-latest"]
+"python-version" = ["3.6", "3.7", "3.8", "3.9"]
+
+# 4. "jobs.build" should have "runs-on" with value "${{ matrix.os }}"
+[".github/workflows/python.yaml".jobs.build]
+"runs-on" = "${{ matrix.os }}"
+
+# 5. "{{" and "}}" will conflict with Jinja https://github.com/andreoliwa/nitpick/issues/283
+# So we need a way to turn on/off Jinja templating.
+# Probably "false" will be the default, to keep compatibility.
+# Whoever wants to use Jinja will need to set "true" either here or as a global config on .nitpick.toml
+__jinja = false
+
+# 6. Another way to turn off Jinja for a specific key only, not the whole dict
+# (using the "__" syntax from Django filters, SQLAlchemy, factoryboy...)
+"runs-on__no_jinja" = "${{ matrix.os }}"
+
+# 7. Same effect as items 1, 2 and 3, but with a different syntax.
+# 8. Everything that is not a dunder key ("__") will be a dict to be enforced.
+[[".github/workflows/python.yaml".contains]]
```

Comment:
This approach looks closer to the existing API, and that's nice.
It still feels somewhat clunky to have the`__jmespath` attribute, but only mildly.

Reply:
It looks weird indeed.

The dunder names were an attempt to avoid having reserved words.
So we can actually have a key `jmespath` in some config file:

```
[["some/config/file.yaml".contains]]
__jmespath = "path.to.inner.key"
jmespath = "people[].[name, age, height]"
```

The config file:
```yaml
path:
  to:
    inner:
      key:
        jmespath: "people[].[name, age, height]"
```

Highly unlikely situation, though.

---

I guess we can start with a reserved word `jmespath` (without `__` in the beginning), and think about it later when a collision actually happens.